pkg/ioutils/concat.go

```diff
@@ -0,0 +1,123 @@
+package ioutils
+
+import (
+	"io"
+	"errors"
+)
+
+func min(x, y int) int {
+	if x < y {
+		return x
+	}
+	return y
+}
+
+func max(x, y int) int {
+	if x < y {
+		return y
+	}
+	return x
+}
```

Comment:
Because Go is stupid and doesn't have generics. The math package only defines `min`/`max` functions for `float64` types...